recipes/halide/run_test.sh

```diff
@@ -0,0 +1,10 @@
+if [[ $(uname) == "Darwin" ]]; then
+    # don't use clang++ from the env
+    export CXX=c++
```

Comment:
toolchain sets CXX=clang++, which normally resolves to /usr/bin/clang++. When clangdev is installed in conda, the version in the env will be used instead of the version toolchain normally expects. This doesn't work (I haven't investigated why, but the conda-packaged clang can't build the library). An equivalent fix would be setting it to /usr/bin/clang++.

If toolchain set `export CXX=/usr/bin/clang++` so that it always pointed to the same compiler, this wouldn't be necessary.